desktop/sources/scripts/library.js

```diff
@@ -287,29 +287,62 @@ function Library (ronin) {
   // Pixels
 
   this.pixels = (rect, fn, q) => {
-    const img = ronin.surface.context.getImageData(0, 0, rect.w, rect.h)
+    const img = ronin.surface.context.getImageData(rect.x, rect.y, rect.w, rect.h)
     for (let i = 0, loop = img.data.length; i < loop; i += 4) {
       const pixel = { r: img.data[i], g: img.data[i + 1], b: img.data[i + 2], a: img.data[i + 3] }
       const processed = fn(pixel, q)
-      img.data[i] = processed[0]
-      img.data[i + 1] = processed[1]
-      img.data[i + 2] = processed[2]
-      img.data[i + 3] = processed[3]
+      img.data.set(processed, i);
     }
-    ronin.surface.context.putImageData(img, 0, 0)
+    ronin.surface.context.putImageData(img, rect.x, rect.y)
     return rect
   }
 
   this.saturation = (pixel, q = 1) => {
-    const color = 0.2126 * pixel.r + 0.7152 * pixel.g + 0.0722 * pixel.b
-    return [(color * (1 - q)) + (pixel.r * q), (color * (1 - q)) + (pixel.g * q), (color * (1 - q)) + (pixel.b * q), pixel.a]
+    const color = 0.2126 * pixel.r + 0.7152 * pixel.g + 0.0722 * pixel.b * (1 - q)
```

Comment:
not sure about the saturation change, this * (1 - q) multiplication is only on the 3rd color

Reply:
I'll correct

Reply:
oh oops, sorry about that ^_^